### PR TITLE
*: add tests for config sturcts

### PR DESCRIPTION
**Pull request by overvenus**
Add missing tests for those structs.

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -111,7 +216,7 @@ macro_rules! build_cf_opt {
     }};
 }
 
-cf_config!(DefaultCfConfig);
+cf_config!(DefaultCfConfig, test_default_cf_config);
```
Comment (siddontang) [inline src/config.rs]: can we use a cf_config_test macro instead? 
It is strange to put them together in one macro.

**Comment by siddontang** [comment]
PTAL @BusyJay

File: src/config.rs [hunk 2]
```diff
@@ -690,3 +690,457 @@ impl TiKvConfig {
...
+    use toml::{self, ser};
+
+    extern crate serde_test;
+    use self::serde_test::{assert_de_tokens, Token};
+
+    #[test]
+    fn test_toml_serde_roundtrippping() {
+        {
+            let value = DbConfig::default();
+            let dump = ser::to_string_pretty(&value).unwrap();
```
Comment (BusyJay) [inline src/config.rs]: Why not use `toml::to_string_pretty`?

File: src/config.rs [hunk 3]
```diff
@@ -690,3 +690,457 @@ impl TiKvConfig {
...
+            assert_eq!(value, load);
+        }
+        {
+            let value = RaftDefaultCfConfig::default();
+            let dump = ser::to_string_pretty(&value).unwrap();
+            let load = toml::from_str(&dump).unwrap();
+            assert_eq!(value, load);
+        }
+        {
+            let value = TiKvConfig::default();
```
Comment (BusyJay) [inline src/config.rs]: I think this has covered all the other cases.

Comment (overvenus) [reply src/config.rs]: Yes. I think it ok to unit test all of them, these tests are fast. And if the tests fails, we can find out which one breaks it.

Comment (BusyJay) [reply src/config.rs]: Actually if they are not equal, the text will be printed out.

File: src/storage/config.rs [hunk 4]
```diff
@@ -61,3 +61,50 @@ impl Config {
...
+                len: 7,
+            },
+
+            Token::Str("data_dir"),
+            Token::Str(DEFAULT_DATA_DIR),
+
+            Token::Str("gc_ratio_threshold"),
+            Token::F64(config.gc_ratio_threshold),
+
+            Token::Str("scheduler_notify_capacity"),
```
Comment (BusyJay) [inline src/storage/config.rs]: I think it should be `scheduler-notify-capacity`. It seems that the case is useless.

**Comment by overvenus** [comment]
PTAL

File: src/config/mod.rs [hunk 5]
```diff
@@ -690,3 +690,290 @@ impl TiKvConfig {
...
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use toml;
+
+    #[test]
+    fn test_toml_serde_roundtrippping() {
```
Comment (BusyJay) [inline src/config/mod.rs]: Weird name. I think `test_toml_serde` is enough.

**Pull request merged**